python_modules/libraries/dagster-census/dagster_census/components/census_component.py

```diff
@@ -0,0 +1,158 @@
+    ResolvedDefsStateConfig,
+)
+from dagster_shared import check
+from dagster_shared.serdes.serdes import deserialize_value
+
+from dagster_census.components.census_scaffolder import CensusComponentScaffolder
+from dagster_census.resources import CensusResource
+from dagster_census.translator import (
+    CensusMetadataSet,
+    CensusSync,
+    CensusWorkspaceData,
+    DagsterCensusTranslator,
+)
+
+
+class CensusSyncSelectorByName(dg.Model):
+    by_name: Annotated[
+        Sequence[str],
+        pydantic.Field(..., description="A list of sync names to include in the collection."),
+    ]
+
+
+class CensusSyncSelectorById(dg.Model):
+    by_id: Annotated[
+        Sequence[int],
+        pydantic.Field(..., description="A list of sync IDs to include in the collection."),
+    ]
+
+
+def resolve_sync_selector(
+    context: dg.ResolutionContext, model
+) -> Optional[Callable[[CensusSync], bool]]:
+    if isinstance(model, str):
+        model = context.resolve_value(model)
+
+    if isinstance(model, CensusSyncSelectorByName):
+        return lambda sync: sync.name in model.by_name
+    elif isinstance(model, CensusSyncSelectorById):
+        return lambda sync: sync.id in model.by_id
+    else:
+        check.failed(f"Unknown connection target type: {type(model)}")
+
+
+@public
+@dataclass
+@dg.scaffold_with(CensusComponentScaffolder)
+class CensusComponent(StateBackedComponent, dg.Resolvable):
+    """Loads Census syncs from a Census workspace as Dagster assets.
+    Materializing these assets will trigger the Census sync, enabling
+    you to schedule Census syncs using Dagster.
+
+    Example:
+
+        .. code-block:: yaml
+
+            # defs.yaml
+
+            type: dagster_census.CensusComponent
+            attributes:
+              census_resource:
+                api_key: "{{ env.CENSUS_API_KEY }}"
+              sync_selector:
+                by_name:
+                  - my_first_sync
+                  - my_second_sync
+    """
+
+    census_resource: Annotated[
+        CensusResource,
+        dg.Resolver(
+            lambda context, model: CensusResource(**resolve_fields(model, CensusResource, context))
+        ),
+    ]
+
+    sync_selector: Annotated[
+        Optional[Callable[[CensusSync], bool]],
+        dg.Resolver(
+            resolve_sync_selector,
+            model_field_type=Union[str, CensusSyncSelectorByName, CensusSyncSelectorById],
+            description="Function used to select Census syncs to pull into Dagster.",
+        ),
+    ] = None
+
+    defs_state: ResolvedDefsStateConfig = field(
+        default_factory=DefsStateConfigArgs.legacy_code_server_snapshots
+    )
+
+    @property
+    def defs_state_config(self) -> DefsStateConfig:
+        default_key = f"{self.__class__.__name__}"
+        return DefsStateConfig.from_args(self.defs_state, default_key=default_key)
+
+    @cached_property
+    def translator(self) -> DagsterCensusTranslator:
+        return DagsterCensusTranslator()
```

Comment:
let's remove the layer of indirection here -- for net-new integrations, we don't actually need a `XTranslator` class, and can instead just put a `get_asset_spec` method on the component class itself (you can check out the `OmniComponent` as a reference for that.